esmvaltool/_recipe.py

```diff
@@ -832,6 +832,10 @@ def _initialize_variables(self, raw_variable, raw_datasets):
             if ('cmor_table' not in variable
                     and variable.get('project') in CMOR_TABLES):
                 variable['cmor_table'] = variable['project']
+            if 'end_year' in variable and 'max_years' in self._cfg:
+                variable['end_year'] = min(
+                    variable['end_year'],
+                    variable['start_year'] + self._cfg['max_years'] - 1)
```

Comment:
Don't we need a check on `max_years > 0`?

Reply:
Sure, I can add that.